packages/server/src/sdk/app/applications/general.ts

```diff
@@ -0,0 +1,19 @@
+const URL_REGEX_SLASH = /\/|\\/g
+
+export function getAppUrl(ctx: {
+  request: { body: { name?: string; url?: string } }
```

Comment:
There's a pattern introduced here: https://github.com/Budibase/budibase/pull/8883/files#diff-8e575ce9e55cd6e844436a203115309c30a986350847a75c080c5735ae526ada

For defining the body type using generics. This can be ported here to allow for saying

```ts
export function getAppUrl(ctx: BBContext<SomeAppType>) { }
```
Where `ctx.request.body` will then have the fields from the types available. 

However I don't think that's what we should do here, the `SomeAppType` doesn't really exist, it's partial fields from:
- POST /api/applications
- PUT /api/applications/:appId
- Constructing the ctx for the migration


As a general rule I think It's best we keep `ctx` out of the sdk so it's framework agnostic and typed as much as possible. Instead changing the method signature here to:

```ts
export function getAppUrl(opts: { name?: string, url?: string) { }
```

We could then even remove the manual construction of context from the migration, which points to this being the right way to go